### PR TITLE
Support AUTH_TYPE_CERT_PEM and AUTH_TYPE_NONE

### PR DESCRIPTION
**Pull request by pem70**
**What It Does**
Supports for `AUTH_TYPE_CERT_PEM` and `AUTH_TYPE_NONE` in `session` addressed in [#291] and [#296]

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/core/zowe/core_for_zowe_sdk/session.py [hunk 1]
```diff
@@ -64,9 +66,14 @@ def __init__(self, props: dict) -> None:
         elif props.get("tokenValue") is not None:
             self.session.tokenValue = props.get("tokenValue")
             self.session.type = session_constants.AUTH_TYPE_BEARER
+        elif props.get("cert") is not None and props.get("certKey") is not None:
+            self.session.cert = props.get("cert")
+            self.session.certKey = props.get("certKey")
```
Comment (zFernand0) [inline src/core/zowe/core_for_zowe_sdk/session.py:71]: I believe the properties we would want to use are `certFile` and `certKeyFile`, but only here.
I think it's fine to keep it as `cert` and `certKey` in the `ISession` object 😋

**Review by zFernand0 — CHANGES_REQUESTED** [review]
TL;DR: Changes LGTM! 😋 

---

After testing the AUTH_TYPE_NONE, I'm wondering if we should update the sample SDK. I was able to use some of that code to test this functionality and it works! 🥳 
```python
from zowe.core_for_zowe_sdk.sdk_api import SdkApi

sample = SdkApi({"host": "jsonplaceholder.typicode.com"}, "/")
custom_args = sample._create_custom_request_arguments()
custom_args["json"] = {"title": "foo", "body": "bar", "userId": 10}
custom_args["url"] = f"{sample._request_endpoint}/posts"
response_json = sample.request_handler.perform_request("POST", custom_args, expected_code=[201])
print(response_json)
```

---

Tested with a local instance of the APIML (using [this instructions](https://github.com/zowe/zowe-explorer-vscode/pull/2802#pullrequestreview-2044034545)) and...

I can't seems to be able to use certificates as a valid authentication mechanism.
😢 

While trying to resolve this, I found a few other places where we may need to make changes in order to pass certs properly to the `requests.Session()` object.
And keep in mind that we would also want to make sure not to validate certs if rejectUnauthorized is False. 😋

File: src/core/zowe/core_for_zowe_sdk/session.py [hunk 2]
```diff
@@ -64,15 +65,24 @@ def __init__(self, props: dict) -> None:
...
+            self.session.type = session_constants.AUTH_TYPE_NONE
+            self.__logger.info("Authentication method not supplied")
+            # raise Exception("An authentication method must be supplied")
 
         # set additional parameters
         self.session.basePath = props.get("basePath")
         self.session.port = props.get("port", self.session.port)
         self.session.protocol = props.get("protocol", self.session.protocol)
-        self.session.rejectUnauthorized = props.get("rejectUnauthorized", self.session.rejectUnauthorized)
+        self.session.rejectUnauthorized = False if props.get("rejectUnauthorized") == "False" else True
```
Comment (zFernand0) [inline src/core/zowe/core_for_zowe_sdk/session.py:85]: I believe the comparison needs to be with a boolean and not with a string 😋 
```suggestion
        self.session.rejectUnauthorized = False if props.get("rejectUnauthorized") == False else True
```

**Review by zFernand0 — CHANGES_REQUESTED** [review]
Left a small comment, but otherwise, the functionality LGTM! 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋
